darts/timeseries.py

```diff
@@ -929,8 +957,21 @@ def _combine_from_pd_ops(self, other: 'TimeSeries',
         raise_if_not(self.has_same_time_as(other), 'The two TimeSeries must have the same time index.', logger)
         raise_if_not(self._df.shape == other._df.shape, 'The two TimeSeries must have the same shape.', logger)
 
-        series = combine_fn(self._df, other.pd_dataframe())
-        return TimeSeries(series, self.freq())
+        df_a = self._df
+        df_b = other._df
```

Comment:
That's probably a little faster than calling `other.pd_dataframe()`, but we now rely on `combine_fn` having no side effect on the dataframes it receives in arguments. I think that's OK in this context, given that _combine_from_pd_ops() is private but worth keeping in mind..